hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java

```diff
@@ -224,6 +224,8 @@ public static void initClusterAndEnv() throws IOException,
           bucketLayout.name());
     } else {
       bucketLayout = BucketLayout.LEGACY;
+      conf.set(OzoneConfigKeys.OZONE_FS_DEFAULT_BUCKET_LAYOUT,
```

Comment:
Can we add specific test case safe guarding ofs bucket creation behavior. Probably you can add a new test class itself to verify different cases:

Pre-requisite) ozone.default.bucket.layout Server defaults to OBJECT_STORE. As this should be ignored at any point of time.

case-1) client side configured to empty, use default FSO. Should create buck as FSO.
case-2) client side configured to OBS, throws Exception
case-3) client side configured to LEGACY, ofs should create buck as LEGACY.
case-4) client side configured to FSO, ofs should create buck as FSO.

Reply:
Added the tests.